bigtable/tests/system.py

```diff
@@ -272,6 +272,88 @@ def test_delete_column_family(self):
         # Make sure we have successfully deleted it.
         self.assertEqual(temp_table.list_column_families(), {})
 
+    def test_retry(self):
+        import subprocess, os, stat, platform, ssl
+        from google.cloud.bigtable.client import Client
+        from google.cloud.bigtable.instance import Instance
+        from google.cloud.bigtable.table import Table
+
+        # import for urlopen based on version
+        try:
+            # python 3
+            from urllib.request import urlopen
+        except ImportError:
+            # python 2
+            from urllib2 import urlopen
+
+        TEST_SCRIPT = 'tests/retry_test_script.txt'
+        SERVER_NAME = 'retry_server'
+        SERVER_ZIP = SERVER_NAME + ".tar.gz"
+
+        def download_server():
+            # Download server
+            MOCK_SERVER_URLS = {
+                'Linux': 'https://storage.googleapis.com/cloud-bigtable-test/retries/retry_server_linux.tar.gz',
+                'Darwin': 'https://storage.googleapis.com/cloud-bigtable-test/retries/retry_server_mac.tar.gz',
+            }
+
+            test_platform = platform.system()
+            if test_platform not in MOCK_SERVER_URLS:
+                self.skip('Retry server not available for platform {0}.'.format(test_platform))
+
+            context = ssl._create_unverified_context()
+            mock_server_download = urlopen(MOCK_SERVER_URLS[test_platform], context=context).read()
+            mock_server_file = open(SERVER_ZIP, 'wb')
+            mock_server_file.write(mock_server_download)
+
+            # Unzip server
+            subprocess.call(['tar', 'zxvf', SERVER_ZIP, '-C', '.'])
+            os.remove(SERVER_ZIP)
+
+        def process_scan(table, range, ids):
+            range_chunks = range.split(",")
+            range_open = range_chunks[0].lstrip("[")
+            range_close = range_chunks[1].rstrip(")")
+            rows = table.read_rows(range_open, range_close)
+            rows.consume_all()
+
+        if not os.environ.get("DO_NOT_DOWNLOAD_TEST_SERVER"):
```

Comment:
How about:

```python
if os.environ.get('DOWNLOAD_BIGTABLE_TEST_SERVER') == '1':
    download_server()
```

Reply:
Well, then developers running these tests on their dev machines would have to set an environment variable manually before they ran the system tests - do we want that?

Reply:
OK, then we can make it default to download it iff:

* the env var is unspecified, OR
* it's specified to `1` (since the Circle CI explicitly defines it to `0`, it will disable it)

```python
should_download = os.environ.get('DOWNLOAD_BIGTABLE_TEST_SERVER')
if should_download is None or should_download == '1':
    download_server()
```

Reply:
Done